### PR TITLE
Install matching gdb xmethod next to libsycl.so

### PR DESCRIPTION
**Pull request by mihails-strasuns-intel**
Small tweak to https://github.com/intel/llvm/pull/821

Original installation layout required explicit loading of the python
module via user-configured .gdbinit, which has been decided sub-optimal.

New naming/placement will allow to make use of
https://sourceware.org/gdb/current/onlinedocs/gdb/Python-Auto_002dloading.html

### CODE REVIEW

File: sycl/CMakeLists.txt [hunk 1]
```diff
@@ -152,10 +151,7 @@ COMMENT "Copying SYCL headers ...")
 install(DIRECTORY "${sycl_inc_dir}/." DESTINATION "${LLVM_INST_INC_DIRECTORY}" COMPONENT sycl-headers)
 
 # Auxilliary extras for SYCL headers
-add_custom_target(sycl-headers-extras ALL
-COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_CURRENT_SOURCE_DIR}/xmethods ${dst_dir}
-COMMENT "Copying SYCL header xmethod scripts ...")
-install(DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/xmethods" DESTINATION ${LLVM_PYTHON_DIRECTORY} COMPONENT sycl-headers-extras)
+install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/xmethods/libsycl.so-gdb.py" DESTINATION lib COMPONENT sycl-headers-extras)
```
Comment (mihails-strasuns-intel) [inline sycl/CMakeLists.txt]: I am not sure if it is possible to refer to `libsycl.so` name via cmake variable in this context.

Comment (alexbatashev) [reply sycl/CMakeLists.txt]: @mihails-strasuns-intel Are you looking for https://cmake.org/cmake/help/latest/command/get_target_property.html and https://cmake.org/cmake/help/latest/prop_tgt/LIBRARY_OUTPUT_NAME.html ?

Comment (mihails-strasuns-intel) [reply sycl/CMakeLists.txt]: That didn't work (looks like it doesn't resolve default names), but I have found `$<TARGET_FILE_NAME:sycl>` which did the trick.

Updated (also moved the definition down to the place where `sycl` target is already defined).

**Comment by mihails-strasuns-intel** [comment]
Update:
- disable installation on Windows

**Pull request merged**